docker/build/devdeps.Dockerfile

```diff
@@ -18,115 +18,90 @@
-    # using the same toolchain as CUDA-Q by default.
-    && source /opt/llvm/bootstrap/init_command.sh \
-    && bash /scripts/install_prerequisites.sh \
-    && apt-get remove -y ca-certificates \
-    && apt-get autoremove -y --purge && apt-get clean && rm -rf /var/lib/apt/lists/*
-
-# Pre-built binaries for doxygen are only available for x86_64.
-FROM ${base_image} as doxygenbuild
-RUN if [ "$(uname -m)" != "x86_64" ]; then \
-        apt-get update && apt-get install -y wget unzip make cmake flex bison gcc g++ python3 \
-        # Fixed commit corresponding to release 1.9.7
-        && wget https://github.com/doxygen/doxygen/archive/6a2ce4d18b5af1ca501bcf585e4c8e2b2b353b0f.zip -q -O repo.zip \
-        && unzip repo.zip && mv doxygen* repo && rm repo.zip \
-        && cmake -G "Unix Makefiles" repo && cmake --build . --target install --config Release \
-        && rm -rf repo && apt-get remove -y wget unzip make cmake flex bison gcc g++ python3 \
-        && apt-get autoremove -y --purge && apt-get clean && rm -rf /var/lib/apt/lists/*; \
+        wget git unzip \
+        python3-dev python3-pip && \
+    python3 -m pip install --no-cache-dir numpy && \
+    apt-get autoremove -y --purge && apt-get clean && rm -rf /var/lib/apt/lists/*
+ADD scripts/install_toolchain.sh /cuda-quantum/scripts/install_toolchain.sh
+RUN source /cuda-quantum/scripts/install_toolchain.sh \
+        -e "$LLVM_INSTALL_PREFIX/bootstrap" -t ${toolchain}
+
+## [Source Dependencies]
+ADD scripts/install_prerequisites.sh /cuda-quantum/scripts/install_prerequisites.sh
+ADD scripts/build_llvm.sh /cuda-quantum/scripts/build_llvm.sh
+ADD cmake/caches/LLVM.cmake /cuda-quantum/cmake/caches/LLVM.cmake
+ADD tpls/customizations/llvm /cuda-quantum/tpls/customizations/llvm
+ADD .gitmodules /cuda-quantum/.gitmodules
+ADD .git/modules/tpls/pybind11/HEAD /.git_modules/tpls/pybind11/HEAD
+ADD .git/modules/tpls/llvm/HEAD /.git_modules/tpls/llvm/HEAD
+
+# This is a hack so that we do not need to rebuild the prerequisites 
+# whenever we pick up a new CUDA-Q commit (which is always in CI).
+RUN cd /cuda-quantum && git init && \
+    git config -f .gitmodules --get-regexp '^submodule\..*\.path$' | \
+    while read path_key local_path; do \
+        if [ -f "/.git_modules/$local_path/HEAD" ]; then \
+            url_key=$(echo $path_key | sed 's/\.path/.url/') && \
+            url=$(git config -f .gitmodules --get "$url_key") && \
+            git update-index --add --cacheinfo 160000 \
+            $(cat /.git_modules/$local_path/HEAD) $local_path; \
+        fi; \
+    done && git submodule init && git submodule
+# Build compiler-rt (only) since it is needed for code coverage tools
+RUN LLVM_PROJECTS='clang;lld;mlir;python-bindings;compiler-rt' \
+    bash /cuda-quantum/scripts/install_prerequisites.sh -t ${toolchain}
+
+## [Dev Dependencies]
+RUN if [ "$(uname -m)" == "x86_64" ]; then \
+        # Pre-built binaries for doxygen are (only) available for x86_64.
```

Comment:
Why do we need to build documentation on multiple platforms by default?

Reply:
We don't as far as our pipelines are concerned (we only build docs on x86_64). I still kept this in the dev images, so that people who work on Mac have a dev image where they can build the docs as well locally.